### PR TITLE
[ARO-12459] Deduplicate our dependencies on go-jose to a single version

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Fixes ARO-12459

### What this PR does / why we need it:

Move to go-jose v4 instead of the gopkg.in version  + go-jose v3.

### Test plan for issue:

E2E, unit tests

### Is there any documentation that needs to be updated for this PR?

N/A

### How do you know this will function as expected in production? 
E2E

### CODE REVIEW

File: test/util/error/error.go [hunk 1]
```diff
@@ -1,21 +1,47 @@
...
+
+const trimWhitespace assertOptions = 1
+
+func TrimWhitespace() assertOptions {
+	return trimWhitespace
+}
 
 // AssertErrorMessage asserts that err.Error() is equal to wantMsg.
-func AssertErrorMessage(t *testing.T, err error, wantMsg string) {
+func AssertErrorMessage(t *testing.T, err error, wantMsg string, opts ...assertOptions) {
```
Comment (rajdeepc2792) [inline test/util/error/error.go:21]: Can we introduce unit tests for the function?

Comment (hawkowl) [reply test/util/error/error.go:21]: It's a bit hard when it's testing.T code. I did verify that it works as I was using it to help debug the issue :)

**Review by jaitaiwan — APPROVED** [review]
LGTM

**Pull request merged**